src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -119,7 +125,13 @@ public PatternHelper( final ItemStack is, final World w )
 
 			for( int x = 0; x < outTag.tagCount(); x++ )
 			{
-				final ItemStack gs = new ItemStack( outTag.getCompoundTagAt( x ) );
+				NBTTagCompound resultItemTag = outTag.getCompoundTagAt( x );
+				final ItemStack gs = new ItemStack( resultItemTag );
+
+				if (!resultItemTag.hasNoTags() && gs.isEmpty())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)